src/extensions/animation/controls.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * Internal
+ */
+import { animationTypes } from './animation-types';
+
+/**
+ * External dependencies
+ */
+import _ from 'lodash';
+import classnames from 'classnames';
+import { AnimationIcon } from '@godaddy-wordpress/coblocks-icons';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { BlockControls, BlockPreview } from '@wordpress/block-editor';
+import {
+	DropdownMenu,
+	MenuItem,
+	MenuGroup,
+	Popover,
+	Toolbar,
+	Tip
+} from '@wordpress/components';
+import { check } from '@wordpress/icons';
+
+function PreviewAnimationPopover( { hoveredAnimation, selected } ) {
+	if ( ! hoveredAnimation ) return null;
+
+	const block = _.cloneDeep(selected);
+
+	block.attributes.animation = hoveredAnimation;
+
+	return (
+		<div className="block-editor-block-switcher__popover__preview__parent">
+			<div className="block-editor-block-switcher__popover__preview__container">
+				<Popover
+					className="block-editor-block-switcher__preview__popover"
+					position="bottom right"
+					focusOnMount={ false }
+				>
+					<div className="block-editor-block-switcher__preview coblocks__preview">
+						<div className="block-editor-block-switcher__preview-title">
+							{ __( 'Preview animation' ) }
+						</div>
+						<BlockPreview
+							autoHeight
+							blocks={ { ...block } }
+						/>
+						<Tip>
+							{ __( 'Animations are applied as this block comes into view on the page.', 'coblocks' ) }
+						</Tip>
+					</div>
+				</Popover>
+			</div>
+		</div>
+	);
+}
+
+class Controls extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			hoveredAnimation: null
+		};
+	}
+
+	onChangeHoveredAnimation = (animation) => {
+		this.setState({
+			hoveredAnimation: animation
+		});
+	}
+
+	onAnimationClick = ( onClose, animationClass = null ) => {
+		const { setAttributes } = this.props;
+
+		return () => {
```

Comment:
@jasonlemay do we need to return a function here? Or just the object would be enough?